### ISSUE TITLE
Share button in URL bar color after 1 click wrong color

### ISSUE THREAD

**Issue by lmarceau** [mozilla-mobile/firefox-ios#15023]
## Steps to reproduce
- Navigate to a website
- Click on the share button once
- Observe the share button color not coming back to its original color

### Expected behavior
Share button color is back to its unselected color.

### Actual behavior
Share button color seems to be clicked still

### Device & build information
* Device: iPhone 14 simulator
* Build version: 020dbda5289d87c8fda1ae0cc0e91d7e1befd910
* First seen version: 115
### Notes
Attachments: 

Before any click on the button
![Screenshot 2023-06-16 at 1 36 02 PM](https://github.com/mozilla-mobile/firefox-ios/assets/11338480/017bea67-6749-4df7-a30d-d1f19bee65c6)

After one click on the button, it stays like this
![Screenshot 2023-06-16 at 1 36 10 PM](https://github.com/mozilla-mobile/firefox-ios/assets/11338480/500a8e79-289f-45dc-a9f8-ea4598ac0e60)

**Comment by rajparad** [comment]
Hello @lmarceau , I checked in iPhone 14 in iOS 17.2. but it's not showing me share button. I am new to open source contribution. let me know if I am missing something :).

<img width="405" alt="image" src="https://github.com/mozilla-mobile/firefox-ios/assets/59967463/009b7bef-af05-439d-b306-7cc02109f103">

**Comment by lmarceau** [comment]
@rajparad Hello! You're right, this button change is protected behind a feature flag (as it's an experiment). What you can do to be able to see it is change the following code under `TabLocationView.swift`
```
    var shouldEnableShareButtonFeature: Bool {
        guard featureFlags.isFeatureEnabled(.shareToolbarChanges, checking: .buildOnly) else {
            return false
        }
        return true
    }
```
to always return true instead. 
```
    var shouldEnableShareButtonFeature: Bool {
        return true
    }
```

Then you'll be able to see the share button in the URL bar! Be sure not to commit those changes thought as we still want this to be experimented on. Let me know if there's anything.

**Comment by rajparad** [comment]
Hello @lmarceau , Thanks for the info.  

After debugging I found that in `ShareButton` class, 
`override open var isHighlighted: Bool {
        didSet {
            self.tintColor = isHighlighted ? selectedTintColor : unselectedTintColor
        }
    }` seems the cause of issue. once the button is pressed isHighlighted gets triggered. based on condition it is changing colour to selectedTintColor. 

once we remove/change this condition above ` self.tintColor = isHighlighted ? unselectedTintColor : selectedTintColor`, it seems to work fine.

https://github.com/mozilla-mobile/firefox-ios/assets/59967463/8389d455-ace6-4e8c-9669-e310aa792b8c

**Comment by lmarceau** [comment]
@rajparad very nice! Do you want to open a PR with the fix? 🙏

**Comment by rajparad** [comment]
@lmarceau , sure I can. but I am not sure about ticket number. as I don't have access for JIRA

**Comment by lmarceau** [comment]
The ticket number can be found if you hover your mouse on the Jira task link as shown in this screenshot:
![Screenshot 2024-02-08 at 4 41 46 PM](https://github.com/mozilla-mobile/firefox-ios/assets/11338480/dbe5862a-4269-470a-90d1-afe50466db48)

. If you don't see no worries, you can open the PR with the Github issue number and we can always change it after. It's a technicality 😄

**Comment by rajparad** [comment]
> The ticket number can be found if you hover your mouse on the Jira task link as shown in this screenshot: ![Screenshot 2024-02-08 at 4 41 46 PM](https://private-user-images.githubusercontent.com/11338480/303490970-dbe5862a-4269-470a-90d1-afe50466db48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0Mjg5MzIsIm5iZiI6MTcwNzQyODYzMiwicGF0aCI6Ii8xMTMzODQ4MC8zMDM0OTA5NzAtZGJlNTg2MmEtNDI2OS00NzBhLTkwZDEtYWZlNTA0NjZkYjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA4VDIxNDM1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwMTUyNmY5NjUzM2IxMTM1MmI3OTMwZWYzYjM5NTI5MGM5NTZhNGM0ODU4NGVjNzcyNGQyZmZmOTQ5Y2U4ZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HFoANFVesNkl4-b4M6im4Yzox3ZQWXojfcUbmI9SE_I)
> 
> If you don't see no worries, you can open the PR with the Github issue number and we can always change it after. It's a technicality 😄

yup, I just realized the about hover after I commented. I will open a PR shortly

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Hi Laurie Marceau We didn’t identify any experiment for making URL share button visible again. Please advise. Thanks in advance!

**Comment by data-sync-user** [comment]
➤ Laurie Marceau commented:

If there’s no experiment available then I think testing can be postponed until there is one, or if that functionality is turned ON to be the default user experience.
